common/src/main/java/com/wynntils/wynn/model/map/MapModel.java

```diff
@@ -154,13 +157,29 @@ private static void loadServices() {
                                 allPois.add(new ServicePoi(location, kind));
                             }
                         } else {
-                            WynntilsMod.error("Unknown service type in services.json: " + service.type);
+                            WynntilsMod.warn("Unknown service type in services.json: " + service.type);
                         }
                     }
 
                     return true;
                 })
                 .build());
+
+        handler.addAndDispatch(new RequestBuilder(SPIRITS_JSON_URL, "maps-spirits")
+                .cacheTo(new File(mapDirectory, "spirits.json"))
+                .useCacheAsBackup()
+                .handleJsonArray(json -> {
+                    Type type = new TypeToken<List<MapLocation>>() {}.getType();
+
+                    List<MapLocation> mapLocations = GSON.fromJson(json, type);
+
+                    for (int i = 0; i < mapLocations.size(); i++) {
```

Comment:
Please use modern style `for(... : ...)`

Reply:
We are using the index to give the POIs special numbers in their name

Reply:
Ah, I see. Sorry.